x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/core/security/transport/ProfileConfigurationsTests.java

```diff
@@ -50,25 +56,134 @@ public void testGetInsecureTransportProfileConfigurations() {
         final Environment env = TestEnvironment.newEnvironment(settings);
         SSLService sslService = new SSLService(env);
         final SslConfiguration defaultConfig = sslService.getSSLConfiguration("xpack.security.transport.ssl");
-        final Map<String, SslConfiguration> profileConfigurations = ProfileConfigurations.get(settings, sslService, defaultConfig);
+        final Map<String, SslConfiguration> profileConfigurations = ProfileConfigurations.get(settings, sslService, true);
         assertThat(profileConfigurations.size(), Matchers.equalTo(2));
         assertThat(profileConfigurations.keySet(), Matchers.containsInAnyOrder("none", "default"));
         assertThat(profileConfigurations.get("none").verificationMode(), Matchers.equalTo(SslVerificationMode.NONE));
         assertThat(profileConfigurations.get("default"), Matchers.sameInstance(defaultConfig));
     }
 
+    public void testTransportAndRemoteClusterSslCanBeEnabledIndependently() {
```

Comment:
nit: a custom profile with a comment here stating that custom profiles are only enabled/disabled with the default profile might help future devs understand some intent

Reply:
Good point. I added a comment to the main code `ProfileConfigurations` for this. I think it is more useful over there. See 8e92bff1c307010c3eec4de23301a2d8b1af94d3